### PR TITLE
[16.01] Merge 15.10 with 16.01

### PR DESCRIPTION
**Pull request by carlfeberhard**
Conflicts manually resolved in:
- `mvc/citation-model.js`
- `framework/middleware/remoteuser.py`
- `galaxy.panels.mako`

All (of the above files) could use a look.

### CODE REVIEW

File: lib/galaxy/web/framework/middleware/remoteuser.py [hunk 1]
```diff
@@ -76,22 +76,12 @@ def __call__( self, environ, start_response ):
...
-                """
-                return self.error( start_response, title, message )
-
-            if not safe_str_cmp(environ.get('HTTP_GX_SECRET'), self.config_secret_header):
+        if path_info.startswith( '/api/' ):
+            # The API handles its own authentication via keys
+            # Check for API key before checking for header
+            return self.app( environ, start_response )
+        elif self.config_secret_header is not None:
+            if not safe_str_cmp(environ.get('HTTP_GX_SECRET', ''), self.config_secret_header):
```
Comment (nsoranzo) [inline lib/galaxy/web/framework/middleware/remoteuser.py:99]: I think that the correct merge here should be:

``` python
        if path_info.startswith( '/api/' ):
            # The API handles its own authentication via keys
            # Check for API key before checking for header
            return self.app( environ, start_response )
        elif self.config_secret_header is not None:
            if environ.get('HTTP_GX_SECRET') is None:
                title = "Access to Galaxy is denied"
                message = """
                Galaxy is configured to authenticate users via an external
                method (such as HTTP authentication in Apache), but
                no shared secret key was provided by the
                upstream (proxy) server.</p>
                <p>Please contact your local Galaxy administrator.  The
                variable <code>remote_user_secret</code> and
                <code>GX_SECRET</code> header must be set before you may
                access Galaxy.
                """
                return self.error( start_response, title, message )

            if not safe_str_cmp(environ.get('HTTP_GX_SECRET'), self.config_secret_header):
```

Comment (carlfeberhard) [inline lib/galaxy/web/framework/middleware/remoteuser.py:99]: Are you sure? I thought that version was your idea: https://github.com/galaxyproject/galaxy/pull/1801#discussion_r54139166

Happy to change it, I guess. 

[EDIT: I think I see the difference now]

**Comment by carlfeberhard** [comment]
@nsoranzo Good catch - that's rather clonal code.

**Comment by dannon** [comment]
Remoteuser looks correct to me now.

**Comment by nsoranzo** [comment]
:+1:

**Comment by dannon** [comment]
It looks like the jquery.migrate change from https://github.com/galaxyproject/galaxy/commit/e2a03646fd7c47a0ead962c9842e74f977dab8eb might have been dropped -- is this handled elsewhere in 16.01 (and forward)?

**Comment by nsoranzo** [comment]
@dannon I think this is not needed because of commit 9a659d85d6896dffa280005e66cc72abc0dcbab1 , but I'm not a JS expert.

**Comment by carlfeberhard** [comment]
You're both correct - after this release, its handled in the 'libs' bundle.

**Comment by dannon** [comment]
Awesome.

**Pull request merged**